### PR TITLE
Try another fix for flaky nav test.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Trying another fix for #35235. Based on [this comment](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/plugins/inner-blocks-allowed-blocks.test.js#L78), I'm thinking it might be the same problem, because we're also trying to click the block appender after closing the global inserter here.

## How has this been tested?

Run e2e tests for `specs/experiments/blocks/navigation.test.js` and check they pass.

Restart e2e tests a few times for this PR to see if the flaky test fails.

## Types of changes

Bug fix (non-breaking change which (hopefully) fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by talldan** [comment]
So did the e2e test catch a bug? Has that been reported anywhere?

**Comment by tellthemachines** [comment]
> So did the e2e test catch a bug? Has that been reported anywhere?

If the behaviour described in the comment I linked above is a bug, and if that's what is causing the flakiness, then I guess so. Though the comment is a few months old, there doesn't seem to be an issue, but I can reproduce the behaviour on trunk by opening the block library and then clicking an appender straightaway. The appender becomes focused, and the library sidebar closes, but quick inserter doesn't open.

I'm not even sure if we're having the same issue here, but the failure screenshots show the quick inserter is _not_ open, and either the inserted block or the appender are focused. The appender is always visible, which means it must be the click action on it that's not working for some reason.

**Comment by gwwar** [comment]
Testing behavior with the Global Inserter, whether or not this fails likely depends on how quickly the sidebar closes. If it's still open and we target the inserter, it'll close the sidebar instead of opening the quick inserter. The proposed changes here may work, but it might be better to fix the following.

https://user-images.githubusercontent.com/1270189/136851951-a920d7f9-65f3-4088-871b-726c2b2a497d.mp4

For other reviewers, here's what a local run looks like, using `npm run test-e2e -- --wordpress-base-url=<local instance> --wordpress-username=<username> --wordpress-password=<password> --puppeteer-interactive specs/experiments/navigation-editor.test.js`

https://user-images.githubusercontent.com/1270189/136851882-b39646cc-69ed-4a5a-9326-848f3d5528a9.mp4

**Comment by tellthemachines** [comment]
> The proposed changes here may work, but it might be better to fix the following.

To be clear, you mean fixing things so the quick inserter opens even if the block library sidebar is still open?

**Comment by talldan** [comment]
I'm sure I recall @draganescu previously fixing the issue shown in that first video, but maybe it was reverted?

edit: possibly https://github.com/WordPress/gutenberg/pull/18379?

**Comment by gwwar** [comment]
> To be clear, you mean fixing things so the quick inserter opens even if the block library sidebar is still open?

💯 probably worth taking a quick look to see if it's reasonable, otherwise I think this change is fine to try. I'll give you a tentative ✅ on this PR so you can land this one if needed.

**Review by gwwar — APPROVED** [review]

**Comment by tellthemachines** [comment]
> I'm sure I recall @draganescu previously fixing the issue shown in that first video, but maybe it was reverted?

I don't think it's quite the same issue; here it's the block library sidebar that hijacks any click on the content area to close itself. It doesn't just happen with quick inserter, but also e.g. when clicking "options" on a block toolbar.

**Comment by tellthemachines** [comment]
So I looked into the library sidebar/inserter issue a bit, and wrote out #35575 with my conclusions. There may be better fixes than the one I described there, and I'm not sure we really want to change how the sidebar works in the library, so will wait for some feedback on that.

In the meantime, I'll go ahead and merge this fix to see if we can at least reduce test flakiness.

**Pull request merged**
